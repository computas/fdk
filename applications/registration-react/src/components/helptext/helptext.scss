@import '../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-reg-helptext {
  background-color: $color6;
  border-radius: 5px;
  font-size: 1.4rem;
  strong {
    font-size: 1.8rem;
  }
  .fdk-reg-helptext-more {
    text-decoration: underline;
    white-space: nowrap;
  }
}

.fdk-obligatorisk {
  font-size: 1.1rem;
  color: #ffffff;
  background-color: #4B555F;
  padding: .1em .6em;
  border-radius: 2px;
  position: relative;
  top: -2px;
}