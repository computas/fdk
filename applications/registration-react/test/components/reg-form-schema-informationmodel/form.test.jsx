import React from 'react';
import { shallow } from 'enzyme';
import FormInformationModel, {
  renderInformationModel
} from '../../../src/components/reg-form-schema-informationmodel/form';
import helptext from '../../fixtures/helptext';

let defaultProps;
let wrapper;

beforeEach(() => {
  const { helptextItems } = helptext;
  defaultProps = {
    helptextItems
  };
  wrapper = shallow(<FormInformationModel {...defaultProps} />);
});

test('should render FormInformationModel, { renderS correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render renderInformationModel correctly', () => {
  wrapper = shallow(renderInformationModel(defaultProps));
  expect(wrapper).toMatchSnapshot();
});
