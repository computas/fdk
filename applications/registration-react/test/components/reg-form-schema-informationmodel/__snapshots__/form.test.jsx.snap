// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormInformationModel, { renderS correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <div
      className="mt-4"
    >
      <div
        className="form-group"
      >
        <Helptext
          helptextItems={
            Object {
              "description": Object {
                "nb": "En eksplisitt referanse til informasjonsmodell
* Benyttes til Ã¥ angi en maskinlesbar referanse til informasjonsmodell.",
              },
              "id": "Dataset_informationModel",
              "shortdesc": Object {
                "nb": "Refereranse til datasettets informasjonsmodell",
              },
              "uri": "http://brreg.no/fdk/fields#Dataset_informationModel",
            }
          }
          required={false}
          title="Informasjonsmodell"
        />
        <FieldArray
          component={[Function]}
          linkLabel="Lenke"
          name="informationModel"
          titleLabel="Tittel"
        />
      </div>
    </div>
  </div>
</form>
`;

exports[`should render renderInformationModel correctly 1`] = `<div />`;
